packages/components/src/AppHeaderBar/AppHeaderBar.component.js

```diff
@@ -1,215 +1,180 @@
-			{props.formgroups ? props.formgroups.map(renderFormGroup) : null}
-			<Button {...props.button}>
-				{props.icon ? (<Icon name={props.icon} />) : null}
-				{props.buttonLabel}
-			</Button>
-		</Navbar.Form>
+		<ActionDropdown
+			bsStyle="link"
+			icon="talend-user-circle"
+			id={id}
+			items={items}
+			label={name}
+			noCaret
+		/>
 	);
 }
-renderForm.propTypes = {
-	form: React.PropTypes.shape(Navbar.Form.propTypes),
-	formgroups: React.PropTypes.arrayOf(
-		React.PropTypes.shape(renderFormGroup.propTypes),
-	),
-	button: React.PropTypes.shape(Button.propTypes),
-	buttonLabel: React.PropTypes.string,
+
+User.propTypes = {
+	id: React.PropTypes.string.isRequired,
+	items: ActionDropdown.propTypes.items,
+	name: ActionDropdown.propTypes.label,
 };
 
-export function renderTypeahead(search, index) {
+function Products({ id, items }) {
 	return (
-		<Navbar.Form pullRight role="search" key={index}>
-			<Typeahead {...search} />
-		</Navbar.Form>
+		<ActionDropdown
+			bsStyle="link"
+			icon="talend-launcher"
+			id={id}
+			items={items}
+			label="Apps"
+			noCaret
+		/>
 	);
 }
-renderTypeahead.propTypes = React.PropTypes.shape(Typeahead.propTypes);
-
-export function renderContent(props, index) {
-	if (props.navs) {
-		return props.navs.map(renderNav);
-	}
-	if (props.forms) {
-		return props.forms.map(renderForm);
-	}
-	if (props.search) {
-		return renderTypeahead(props.search, index);
-	}
-	return null;
-}
-renderContent.propTypes = {
-	navs: React.PropTypes.arrayOf(
-		React.PropTypes.shape(renderNav.propTypes),
-	),
-	forms: React.PropTypes.arrayOf(
-		React.PropTypes.shape(renderForm.propTypes),
-	),
-	search: React.PropTypes.shape(Typeahead.propTypes),
+
+Products.propTypes = {
+	id: React.PropTypes.string.isRequired,
+	items: ActionDropdown.propTypes.items,
 };
 
 /**
  * The top bar is the place where the user finds useful information and tools
  * to describe the application
- * the logo, the name of the application, an optionnal global search,
+ * the logo, the name of the application, an optional global search,
  * few icons for feedback & onboarding and a menu to access logout
  * and profile information.
  * @param {object} props   react props
  */
 function AppHeaderBar(props) {
-	let brandLink;
-	if (props.brandLink) {
-		brandLink = (
-			<Button
-				bsStyle="link"
-				role="link"
-				{...props.brandLink}
-			>
-				{props.app}
-			</Button>
-		);
-	} else {
-		brandLink = (
-			<span>{props.app}</span>
-		);
-	}
 	return (
 		<Navbar fluid fixedTop inverse className={`tc-app-header-bar ${theme['tc-app-header-bar']}`}>
 			<Navbar.Header>
 				<Navbar.Brand>
-					{brandLink}
+					<Logo {...props.logo} />
+					{!props.logo.isFull && '|'}
+					<Brand {...props.brand} />
+					{props.env && '|'}
 				</Navbar.Brand>
 				<Navbar.Toggle />
 			</Navbar.Header>
 			<Navbar.Collapse>
-				{props.content && props.content.map(renderContent)}
+				<Nav>
+					{props.env && (<Environment {...props.env} />)}
```

Comment:
if props.env is false I will have an empty Nav.
Could you please put the props.env condition on the <Nav> ?